src/Orleans/Streams/Internal/StreamImpl.cs

```diff
@@ -132,9 +137,19 @@ public Task<IList<StreamSubscriptionHandle<T>>> GetAllSubscriptionHandles()
 
         internal Task UnsubscribeAsync(StreamSubscriptionHandle<T> handle)
         {
+            CheckExecutingContext("UnsubscribeAsync");
             return GetConsumerInterface().UnsubscribeAsync(handle);
         }
 
+        private void CheckExecutingContext(string caller)
+        {
+            var context = GetStreamProviderRuntime().GetCurrentSchedulingContext();
+            if (context != null && !context.ContextType.Equals(SchedulingContextType.Activation))
+            {
+                throw new OrleansException(String.Format("Attemping to {0} on a stream not from within Grain context.", caller));
```

Comment:
A very good question about client!
For client GetCurrentSchedulingContext returns null. Since in general there is no such concept as SchedulingContext on the client, I did not want to return a fake "ClientSchedulingContext". I think null is the right thing here.
Discovered during testing and that is why added context != null.
Will fix the type and rewrite the error msg.